lib/Resource.js

```diff
@@ -104,7 +115,8 @@ Resource.prototype._associateWith = function(resource) {
     }).pop();
     if (!match) return;
 
-    if (relation.meta.many) {
+    if (Resource._rawRelationHasMany(relation)) {
```

Comment:
Can we keep the variable names consistent between functions? We pass it in here as `relation` and it pops out above as `rawRelation`